File: project1/src/app/mentor/mentor.service.ts
import {Injectable} from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';

import { Mentor } from '../models/mentor.model';
// import { userInfo } from 'os';


const httpOptions = {
  headers: new HttpHeaders({ 'Content-Type': 'application/json' })
};

@Injectable()
export class MentorService {

  constructor(private http:HttpClient) {}

  private mentorUrl = 'http://localhost:8495/mentor';
  //private userUrl = '/api';

  public activeData()
  {
    return this.http.get<Mentor[]>(this.mentorUrl+"/active/true");
  }

  public getByMentor(username)
  {
    return this.http.get<Mentor[]>(this.mentorUrl+"/Username/"+username)
  }
  public updateUser(mentor){
    return this.http.put<Mentor[]>(this.mentorUrl+"/update/"+mentor.mentorId,mentor);
  }

  public searchMentor(mentor){
    return this.http.get<Mentor[]>(this.mentorUrl+"/"+ mentor.email+"/"+mentor.password);
  }

  public getUsers() {
    return this.http.get<Mentor[]>(this.mentorUrl); 
  }

  public deleteUser(user) {
    return this.http.delete(this.mentorUrl + "/"+ user.mentorId);
  }
 

  public createUser(user) {
    return this.http.post<Mentor>(this.mentorUrl, user);
  }

}

File: MentorOnDemand/src/main/java/com/example/demo/MentorService.java
package com.example.demo;

import java.util.List;

public interface MentorService {
	Mentor create(Mentor mentor);
		 
	Mentor delete(Long id);
		 
		 List<Mentor> findAll();
		 
		 Mentor findById(Long mentorId);
		 
		 Mentor update(Mentor mentor);
		 
		 List<Mentor> findByEmailAndPassword(String email,String password);
		 List<Mentor> findByActive(Boolean active);
		 List<Mentor> findByUserName(String userName);
		 
	}
